File: tcpclient/src/test/scala/com/pointr/tcp/sparcle/AppContainerServiceTest.scala
package com.pointr.tcp.sparcle

import org.scalatest.FunSuite

class AppContainerServiceTest extends FunSuite {

  test("AppContainerServiceDriver") {
    val ret = AppContainerServiceDriver.run(Array.empty[String])
    println(ret)
    for (api <- Seq("ExecuteAppCommandResp", "ExecuteAppCommand2Resp","GetAppStatusResp", "ReadNextFromSparcleResp",
      "WriteAppDataToSparcleResp","WriteFeedDataToSparcleResp","WriteAppNativeDataToSparcleResp","Foo")) {
      assert(ret.contains(api),s"Missing response for api=$api")
    }
  }

}

File: tcpclient/src/test/java/com/pointr/tcp/java/sparcle/AppContainerServiceTest.java
package com.pointr.tcp.java.sparcle;

import junit.framework.TestCase;
import scala.Array;

public class AppContainerServiceTest extends TestCase {

  public void testAppContainerService() throws InterruptedException {
        String ret = (String) AppContainerServiceDriver.run();
    System.out.println(ret);
    for (String api : new String[]{"ExecuteAppCommandResp", "ExecuteAppCommand2Resp","GetAppStatusResp", "ReadNextFromSparcleResp",
      "WriteAppDataToSparcleResp","WriteFeedDataToSparcleResp","WriteAppNativeDataToSparcleResp"}) {
      assert ret.contains(api) : "Missing response for api=%s".format(api) ;
    }

  }

}
